src/flint/types/fmpz_vec.pyx

```diff
@@ -0,0 +1,80 @@
+from flint.flintlib.fmpz cimport fmpz_struct, fmpz_set, fmpz_init_set
+from flint.flintlib.flint cimport slong
+from flint.flintlib.fmpz_vec cimport _fmpz_vec_init, _fmpz_vec_clear
+
+from flint.types.fmpz cimport fmpz, any_as_fmpz
+
+cimport libc.stdlib
+
+cdef class fmpz_vec:
+    def __cinit__(self, iterable_or_len, bint double_indirect=False):
+        if isinstance(iterable_or_len, int):
+            self.length = iterable_or_len
+        else:
+            self.length = len(iterable_or_len)
+
+        self.val = _fmpz_vec_init(self.length)
```

Comment:
What happens here if `len` raises an exception?

Do we end up calling `_fmpz_vec_clear` on a NULL pointer?

Reply:
We would indeed. Fixed